RELEASE.md

```diff
@@ -0,0 +1,54 @@
+# Release Process
+
+## Scaffolding a New Release
+
+When planning a spec release a maintainer should:
+1. Create a milestone for the release (e.g. [`Buildpack 0.5`](https://github.com/buildpacks/spec/milestone/9))
+1. Create a release branch from `main` branch with name `<spec>/<version>` (e.g. `buildpack/0.5`) or `extensions/<spec>/version>` for extension specifications.
+1. Update the version of the relevant spec in `README.md` and the specification file (e.g. see commit [791363e](https://github.com/buildpacks/spec/commit/791363e329d22a7a116ac09df6e8c739ef21383e)).
+1. Push the branch to the repo.
+
+
+## Release Planning
+
+Issues that are scheduled for a given release should be added to the milestone. The core team will regularly review and adjust the contents of upcoming milestones at the weekly [core team sync](https://github.com/buildpacks/community#core-team).
+
+## Contributing to a Release
```

Comment:
This could maybe be broken out into a `CONTRIBUTING.md`?

Reply:
I'm :+1: for this suggestion.